### ISSUE TITLE
Default to using resolvers for queries

### ISSUE THREAD

**Issue by radar** [rmosolgo/graphql-ruby#4468]
:wave: Long time user, first time poster.

In the Getting Started Guide, it's suggested to use this syntax for creating a new field on the `QueryType`:

```ruby
class QueryType < GraphQL::Schema::Object
  description "The query root of this schema"

  # First describe the field signature:
  field :post, PostType, "Find a post by ID" do
    argument :id, ID
  end

  # Then provide an implementation:
  def post(id:)
    Post.find(id)
  end
end
```

An alternative way to define this field would be to separate it out into its own class so that the code that exists in `QueryType` would be:

```ruby
class QueryType < GraphQL::Schema::Object
  description "The query root of this schema"

  field :post, resolver: Resolvers::Post
end
```

This syntax aligns with what's suggested in the "Hooking up Mutations" section here: https://graphql-ruby.org/mutations/mutation_classes.html#hooking-up-mutations.

```ruby
class Types::Mutation < Types::BaseObject
  field :create_comment, mutation: Mutations::CreateComment
end
```

Whereas this syntax could be used instead:

```ruby
class Types::Mutation < Types::BaseObject
  field :create_comment, CommentType, null: false

  def create_comment(input)
  end
end
```

(Apologies if that's wrong, I'm writing it off the top my head. But I think you get the idea anyway.)

I believe having this inconsistency between how _queries_ and _mutations_ are defined is, potentially, confusing to new users of this library. 

I would like to propose that the suggested query syntax be shortened to used distinct resolver classes instead of defining the field resolution method directly in the `QueryType` class. I think this works better as any significantly complex `QueryType` class gets quite cumbersome to work with when there are too many method definitions in there. Especially when both the arguments for the `QueryType` fields are complex, as is the resolution for each field.

Can we switch the `QueryType` to use resolvers, as suggested here: https://graphql-ruby.org/fields/resolvers.html#when-do-you-really-need-a-resolver? Making resolvers the _de facto_ way of using this library, rather than defining methods inside of `QueryType`? I would really love to bring these two classes of `QueryType` and `MutationType` into alignment.

**Comment by rmosolgo** [comment]
I'm definitely open to it! Mutations got special treatment in the past in order to accommodate Relay-specific schema requirements, but maybe doing root-level queries would be better this way. Are you interested in taking a crack at updating the generators and docs?

**Comment by rmosolgo** [comment]
Fixed by #4513
